compiler/docgen.nim

```diff
@@ -39,7 +39,7 @@ type
     exampleCounter: int
     emitted: IntSet # we need to track which symbols have been emitted
                     # already. See bug #3655
-    destFile*: AbsoluteFile
+    destFile*: AbsoluteFile # that's a lie...
```

Comment:
Well ok, I believe you. Can we fix it so that it doesn't lie?

Reply:
done (https://github.com/nim-lang/Nim/issues/13121 was root caused; this PR includes a fix for it)